flaml/model.py

```diff
@@ -856,20 +845,9 @@ def predict(self, X, **pred_kwargs):
                 test_dataset,
                 metric_key_prefix="predict",
             )
-
-        if self._task == SEQCLASSIFICATION:
-            return np.argmax(predictions.predictions, axis=1)
-        elif self._task == SEQREGRESSION:
-            return predictions.predictions.reshape((len(predictions.predictions),))
-        elif self._task == TOKENCLASSIFICATION:
-            return np.argmax(predictions.predictions, axis=2)
-        elif self._task == SUMMARIZATION:
-            decoded_preds = self.tokenizer.batch_decode(
-                predictions.predictions, skip_special_tokens=True
-            )
-            return decoded_preds
-        elif self._task == MULTICHOICECLASSIFICATION:
-            return np.argmax(predictions.predictions, axis=1)
+        return postprocess_prediction(

```

Comment:
When the task is NLG, the predictions go through different operations in _compute_metrics_by_dataset_name and in this function.

Reply:
The additional lines of _compute_metrics_by_Dataset_name are for postprocess the labels, and postprocess_seq2seq_prediction is for the rouge score computation.

Reply:
For rogue score computation, why is postprocess_seq2seq_prediction() only needed in _compute_metrics_by_dataset_name()? Why don't you need to do the same when computing the score after the training is finished?